File: ncms-engine/ncms-engine-core/src/main/java/com/softmotions/ncms/marketing/mtt/MttRule.java
package com.softmotions.ncms.marketing.mtt;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonProperty;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import java.io.Serializable;
import java.util.Date;

/**
 * @author Tyutyunkov Vyacheslav (tve@softmotions.com)
 */
@XmlAccessorType(XmlAccessType.NONE)
@JsonAutoDetect(
        creatorVisibility = JsonAutoDetect.Visibility.NONE,
        fieldVisibility = JsonAutoDetect.Visibility.NONE,
        getterVisibility = JsonAutoDetect.Visibility.NONE,
        isGetterVisibility = JsonAutoDetect.Visibility.NONE,
        setterVisibility = JsonAutoDetect.Visibility.NONE
)
public class MttRule implements Serializable {
    @JsonProperty
    private Long id;
    @JsonProperty
    private String name;
    @JsonProperty
    private String description;
    @JsonProperty
    private Long ordinal;
    @JsonProperty
    private Date cdate;
    @JsonProperty
    private Date mdate;
    @JsonProperty
    private boolean enabled = true;
    @JsonProperty
    private long flags;

    public MttRule() {
    }

    public MttRule(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Long getOrdinal() {
        return ordinal;
    }

    public void setOrdinal(Long ordinal) {
        this.ordinal = ordinal;
    }

    public Date getCdate() {
        return cdate;
    }

    public void setCdate(Date cdate) {
        this.cdate = cdate;
    }

    public Date getMdate() {
        return mdate;
    }

    public void setMdate(Date mdate) {
        this.mdate = mdate;
    }

    public boolean getEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public long getFlags() {
        return flags;
    }

    public void setFlags(long flags) {
        this.flags = flags;
    }
}

File: ncms-engine/ncms-engine-core/src/main/java/com/softmotions/ncms/marketing/mtt/MttRule.kt
package com.softmotions.ncms.marketing.mtt

import org.slf4j.LoggerFactory
import java.io.Serializable
import java.util.*

/**
 * Правило фильтрации входящих запросов
 * и контекстное выполнение действий
 *
 * @author Tyutyunkov Vyacheslav (tve@softmotions.com)
 */
data class MttRule(var id: Long = 0,
                   var name: String? = null,
                   var ordinal: Long? = null, // в правилах может быть важен порядок
                   var cdate: Date? = null,
                   var mdate: Date? = null,
                   var flags: Long = 0 // флаги режима работы правила
) : Serializable {

    @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
    constructor(id: java.lang.Long,
                name: java.lang.String,
                ordinal: java.lang.Long,
                cdate: java.sql.Timestamp,
                mdate: java.sql.Timestamp,
                flags: java.lang.Long)
    : this(id.toLong(), name.toString(), ordinal.toLong(), Date(cdate.time), Date(mdate.time), flags.toLong()) {
    }

    companion object {
        private val log = LoggerFactory.getLogger(MttRule::class.java)
    }
}
